### PR TITLE
Minimal PR to enable appveyor support

### PR DESCRIPTION
**Pull request by dsyme**

### CODE REVIEW

**Comment by forki** [comment]
I push it to my fork to see what happens on appveyor. see https://ci.appveyor.com/project/SteffenForkmann/visualfsharp/build/0.0.1.12

**Comment by dsyme** [comment]
It needs some work, am working on it - I've got it enabled in my fork too, e.g.https://ci.appveyor.com/project/dsyme/visualfsharp/build/0.0.1.6.

I created the PR to clear the idea with @latkin, who agrees with the direction in principle as an interim

**Comment by forki** [comment]
@latkin can you please activate this project in appveyor?

**Comment by dsyme** [comment]
Appveyor should now be enabled (@KevinRansom did it)

We have a problem running Perl on the CI server, see https://ci.appveyor.com/project/dsyme/visualfsharp/build/0.0.1.6#L1003, not sure what the error is, perhaps its just not installed (or a wrong version)

Will try the other unit tests now

**Comment by dsyme** [comment]
Great - we have "Waiting to hear about b310e2a — Waiting for AppVeyor build to complete" in this PR now, thanks @KevinRansom  :)

File: build.cmd [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+REM %_msbuildexe% src/fsharp-library-unittests-build.proj /p:UseNugetPackages=true /p:TargetFramework=portable47
+@REM if ERRORLEVEL 1 echo Error: library unittests debug build failed portable47 && goto :eof
+
+REM Dropped for faster build
+REM %_msbuildexe% src/fsharp-library-unittests-build.proj /p:UseNugetPackages=true /p:TargetFramework=portable7
+REM @if ERRORLEVEL 1 echo Error: library unittests debug build failed portable7 && goto :eof
+
+REM Dropped for faster build
+REM %_msbuildexe% src/fsharp-library-unittests-build.proj /p:UseNugetPackages=true /p:TargetFramework=portable78
+REM @if ERRORLEVEL 1 echo Error: library unittests debug build failed portable78 && goto :eof
```
Comment (latkin) [inline build.cmd]: Need profile 259

**Comment by latkin** [comment]
I know time is a factor, but riht now it doesn't look like any portable tests are either compiled or run. The portable stuff is a frequent culprit for build issues, so that would be nice to get some coverage on.

**Comment by ReedCopsey** [comment]
@latkin I agree.  Even having a single portable build would be nice.  Would doing only the most restrictive portable build (259) as a start suffice? 

My experience has always been that it will show most problems that arise in any of the other targets.

**Comment by latkin** [comment]
From looking at candidate build.cmd, seems that some of the profiles are indeed included when building the runtime, I was only speaking of the tests (which suffer from similar build issues on occasion).

**Comment by dsyme** [comment]
AFAIK we have a 30minute window on AppVeyor - I may be wrong (that may be for Travis).  We are pretty close to that.  Let's see where we are, get some of each of the test suites enabled including running all of the core unit tests, and then add stuff.

**Comment by ReedCopsey** [comment]
FYI: You get 40 minutes on AppVeyor, at any level of account.  If we were
to upgrade to paid accounts, the build system would also be quicker, as the
hardware used gets nicer.

**Comment by dsyme** [comment]
Last run we we're on 38min (running just the basic FSharp.Core.Unittests) - 3 test failures for reasons,due to timeouts and one due to a \r\n v. \n difference which surprised me (is AppVeyor checking out the repo with non-standard endings/)

Yes, upgrading would help. It looks like we would need Pro for it to make any real difference.  However the @Microsoft/visualfsharp team's plan is to move to (or augment with) eventually whatever Roslyn use, which is fair enough. Presumably Microsoft will have a paid plan or their own installations for that.

**Comment by dsyme** [comment]
OK, I merged this to enable AppVeyor on commits for the day, please keep an eye on the history https://ci.appveyor.com/project/KevinRansom/visualfsharp-radou/history

**Comment by enricosada** [comment]
perl is the one embedded in git `C:\Program Files (x86)\Git\bin\perl.exe` and is not ok ( [perl version](https://ci.appveyor.com/project/enricosada/visualfsharp/build/0.0.1.2#L840) because msys perl dont have win32 cpan module

**Comment by enricosada** [comment]
@dsyme i think i can fix perl ( download + install in `appveyor-build.cmd` ) if needed.
But there is time limit, is it usefull?

**Comment by latkin** [comment]
@enricosada do you know if msys has support for an alternative module for controlling processes? We could use that if needed.

**Comment by enricosada** [comment]
dont know really, i'll try to install module [Win32](http://search.cpan.org/~jdb/Win32-Process-0.16/ ) from cpan before call perl

**Comment by dsyme** [comment]
@enricosada - yes, it would be useful to fix perl, we can then selectively run subsets of the fsharp and fsharpqa tests

**Comment by enricosada** [comment]
no luck with install Win32 module on perl embedded in msys git 
( http://stackoverflow.com/questions/28283602/use-win32process-with-perl-of-msys-git )

Anyway now with upgraded appveyour we can install activestate perl, i'll try to do that

**Pull request merged**